Source/Urho3D/CSharp/CodeGenerator/CodeGenerator.cs

```diff
@@ -87,6 +90,7 @@ public void Execute(GeneratorExecutionContext context)
 
                         var hasClassName = HasStaticField(typeSymbolInfo, "ClassName");
 
+                        sourceBuilder.AppendLine("[global::Urho3DNet.Preserve(AllMembers=true)]");
```

Comment:
Are you sure source generator generates this stuff for all classes? Even those that get nothing else generated by this generator?

Reply:
It only applied to the classes that inherit from Urho3D Object. I think this is fine since the purpose of this is to allow c# component creation via reflection and the rest of the API is referenced naturally from the code.